starter-code/scripts/models/article.js

```diff
@@ -69,18 +69,31 @@ Article.getAll = function(nextFunction) {
 Article.numWordsAll = function() {
   return Article.allArticles.map(function(currentArticle) {
     return currentArticle.body.match(/\w+/g).length;
-  }).reduce(function() {
-    // TODO: complete this function to sum up all of the words.
-  })
+  }).reduce(function(acc, current) {
+    // TODO: DONE complete this function to sum up all of the words.
+    return acc + current
+  }, 0)
 }
 
 Article.allAuthors = function() {
-  // TODO: return a mapped collection
-      // with just the author names
-
-      //then chain reduce, and set the accumulator to an array
-      // to build a unique list of author names.
+  return Article.allArticles.map(function(allArticles) {
+    return allArticles.author
+  })
+  .reduce(function(acc, current) {
+    if (acc.indexOf(current) < 0) {
+      acc.push(current)
+      return acc
+    }
+    else {
+      return acc
```

Comment:
And, just FYI - if your conditional blocks only contain one statement, you can write them on one line without curly braces, like so:
```
if (acc.indexOf(current) < 0) acc.push(current);
return acc;
```